lib/trino-parquet/src/main/java/io/trino/parquet/ParquetCompressionUtils.java

```diff
@@ -72,9 +72,14 @@ public static Slice decompress(CompressionCodec codec, Slice input, int uncompre
 
     private static Slice decompressSnappy(Slice input, int uncompressedSize)
     {
-        byte[] buffer = new byte[uncompressedSize];
-        decompress(new SnappyDecompressor(), input, 0, input.length(), buffer, 0);
-        return wrappedBuffer(buffer);
+        // Snappy decompressor is more efficient if there's at least a long's worth of extra space
```

Comment:
why?

Reply:
See https://github.com/airlift/aircompressor/blob/master/src/main/java/io/airlift/compress/snappy/SnappyRawDecompressor.java#L78
Code change is copied from OrcSnappyDecompressor